### PR TITLE
Migrate to python3

### PR DESCRIPTION
**Pull request by hidrarga**
I'm proposing to move renew_from_cube.py, which was nothing more than a Yunohost wrapper for the renewal script, and to put it directly inside the [Neutrinet app](https://github.com/Neutrinet/neutrinet_ynh).

So, the main (and only) script is now renew.py, which you can safely run from your computer. If you want to run this script on a cube, there will be a bash script on the Neutrinet app repository. I'm just waiting for Ilja to make his own changes on the packaging stuff. I can also provide the bash script here meanwhile, so that you can review it, but we'll have to move it later on. Let me know what you prefer.

There are quite a lot of changes in this pull request:
- Use pyOpenSSL instead of making system calls to openssl
- Use Argparse, the built-in argument parser in python. I also added a bunch of parameters (see the README.md)
- Use the built-in python logger. By default it only display info (and more critical) messages, but we can choose to display debug messages as well with the `-v` option.
- [Based on this article](https://www.peterbe.com/plog/best-practice-with-retries-with-requests), use a retry session for the queries to the Neutrinet API
- Check that the public part of the client certificate (when provided) will be soon expired before starting the renewal. Before the migration, this task was done by the renew_from_cube.py script, but it's better to move it here, because when you run it on your computer, you'd expect that the script checks the expiration date before renewing.
- Let the user provide the target directory. This part is necessary for the Neutrinet app that will renew the certificates on the cube, because otherwise it would be too difficult to get back the newly generated target directory… Note that when you don't provide the target directory, the directory is generated as before.
- Remove the dependency to StringIO [as it is not needed in python3](https://stackoverflow.com/questions/11914472/stringio-in-python3), and use BytesIO instead.

I successfully tested the script on my machine and on a cube.

### CODE REVIEW

**Comment by tbalthazar** [comment]
Thanks a lot for your contribution! <3
It would be nice to have more context around this. Has it been tested on an Internet Cube yet?
Any plan to update the neutrinet_ynh app that uses this script (at a [specific commit](https://github.com/Neutrinet/neutrinet_ynh/blob/4d8c42a4c3c6173f192fba28e10e7ace40d4e67e/scripts/install#L55))?

I don't have a working cube anymore, maybe @Tharyrok, @datatierce or anybody with more python-foo than I have can confirm it still works?

**Comment by ghost** [comment]
Je te mets des cœurs partout … tu vas finir par croire que je suis amoureux :tongue: 

@hidrarga te répondra bientôt … ils sont deux a travailler sur neutrinet_ynh, le renew_cert **et** de quoi faciliter grandement l'installation des briques. :v:

**Comment by hidrarga** [comment]
> Thanks a lot for your contribution! <3

You are welcome \o/

> It would be nice to have more context around this. Has it been tested on an Internet Cube yet?

True! I updated the description ;) This was tested this morning on a freshly installed cube, but of course that would be better if other people could test it :) 

> Any plan to update the neutrinet_ynh app that uses this script (at a [specific commit](https://github.com/Neutrinet/neutrinet_ynh/blob/4d8c42a4c3c6173f192fba28e10e7ace40d4e67e/scripts/install#L55))?

Yes, Ilja and I are working on it :) But we will first need to merge this pull request before continuing on the Neutrinet app :)

**Comment by tbalthazar** [comment]
Thanks folks!

File: README.md [hunk 1]
```diff
@@ -1,53 +1,60 @@
...
 
-    git clone https://github.com/neutrinet/renew_cert
-    cd renew_cert
-    virtualenv ve
-    source ve/bin/activate
-    pip install -r requirements.txt
+```bash
+git clone https://github.com/neutrinet/renew_cert
+cd renew_cert
+virtualenv ve
```
Comment (Psycojoker) [inline README.md:18]: This won't create a python3 virtualenv I believe :x

Comment (hidrarga) [reply README.md:18]: Indeed, I forgot to change that part in the Readme…

**Pull request merged**